gpt4all-chat/src/chat.cpp

```diff
@@ -124,10 +125,28 @@ void Chat::resetResponseState()
     emit responseStateChanged();
 }
 
-void Chat::prompt(const QString &prompt)
+void Chat::prompt(const QString &prompt, const QList<QUrl> &attachedUrls)
 {
+    newPromptResponsePairInternal(prompt, attachedUrls);
+    emit resetResponseRequested();
+
     resetResponseState();
-    emit promptRequested(m_collections, prompt);
+
+    QStringList attachedContexts;
+    for (const QUrl &url : attachedUrls) {
+        Q_ASSERT(url.isLocalFile());
+        const QString localFilePath = url.toLocalFile();
+        const QFileInfo info(localFilePath);
+        Q_ASSERT(info.suffix() == "xlsx"); // We only support excel right now
+        Q_ASSERT(info.exists());
+        Q_ASSERT(info.isFile());
```

Comment:
This is another TOCTOU. We should check whether XLSXToMD::toMarkdown fails to open the file (via an exception or error status), and then explicitly handle that situation in the same way for debug and release builds.